### ISSUE TITLE
Provide a default exclude list of 'bad' ip ranges

### ISSUE THREAD

**Issue by reedjones** [robertdavidgraham/masscan#638]
Not really a bug or issue but, 

In the readme it says:

> Scanning the entire Internet is bad. For one thing, parts of the Internet react badly to being scanned. For another thing, some sites track scans and add you to a ban list, which will get you firewalled from useful parts of the Internet. Therefore, you want to exclude a lot of ranges.

I'm wondering is there any known comprehensive list of 'bad' ip ranges? 

I've found a few like [this](https://www.hacking-tutorial.com/tips-and-trick/do-not-scan-this-ip-address-ranges/)

Just wondering if there is any kind of official/comprehensive list?

**Comment by GodlyEcstasy** [comment]
Probably not.  What are you trying to avoid?  It appears most of the ranges in the URL you linked are government ranges.

**Comment by reedjones** [comment]
Yeah that's the point. It says in readme that masscan _can_ scan the whole internet but that you _shouldn't_. Trying to void getting on some blacklist for scanning some military network or something ;)

**Comment by reedjones** [comment]
Note- In the nmap documentation it says that they do this (filter out 'bad' ranges) when the iR (random hosts) flag is set.
maybe something to look into

**Comment by dingusgh** [comment]
what IPs does nmap skip?
https://github.com/nmap/nmap/blob/master/libnetutil/netutil.cc#L427:L485

0.0.0.0/8 is IANA Reserved
127.0.0.0/8 Loopback
10.0.0.0/8 is reserved for private nets by RFC1918
169.254.0.0/16 is reserved for DHCP clients seeking addresses RFC3927
172.16.0.0/12 is reserved for private nets by RFC1918
192.0.2.0/24 is reserved for documentation and examples RFC5737
192.168.0.0/16 is reserved for private nets by RFC1918
192.88.99.0/24 is used as 6to4 Relay anycast prefix by RFC3068
198.18.0.0/15 is used for benchmark tests by RFC2544
198.51.100.0/24 is reserved for documentation RFC5737
203.0.113.0/24 is reserved for documentation RFC5737
224.0.0.0/4 is all multicast stuff
240.0.0.0/4 is IANA reserved

notably missing is 100.64.0.0/10 reserved for shared address space RFC6598
maybe I should submit a PR right ;)

see also https://www.iana.org/assignments/ipv4-address-space/ipv4-address-space.xhtml

**Comment by flotwig** [comment]
There's a suggested exclude.conf in the repo itself: https://github.com/robertdavidgraham/masscan/blob/master/data/exclude.conf
